src/mysky/connector.ts

```diff
@@ -0,0 +1,55 @@
+import { Connection, ParentHandshake, WindowMessenger } from "post-me";
+import { createIframe, defaultHandshakeAttemptsInterval, defaultHandshakeMaxAttempts } from "skynet-interface-utils";
+
+import { SkynetClient } from "../client";
+
+export type CustomConnectorOptions = {
+  handshakeMaxAttempts?: number;
+  handshakeAttemptsInterval?: number;
+};
+
+const defaultConnectorOptions = {
+  handshakeMaxAttempts: defaultHandshakeMaxAttempts,
+  handshakeAttemptsInterval: defaultHandshakeAttemptsInterval,
+};
+
+export class Connector {
+  constructor(
+    public url: string,
+    public client: SkynetClient,
+    public childFrame: HTMLIFrameElement,
+    public connection: Connection,
+    public options: CustomConnectorOptions
+  ) {}
+
+  // Static initializer
+
+  static async init(client: SkynetClient, domain: string, customOptions?: CustomConnectorOptions): Promise<Connector> {
+    const opts = { ...defaultConnectorOptions, ...customOptions };
+
+    // Get the URL for the domain on the current portal.
+    const domainUrl = await client.getFullDomainUrl(domain);
+
+    // Create the iframe.
+
```

Comment:
extra space

Reply:
Can we revisit style nits after the hackathon?